__tests__/app.test.js

```diff
@@ -167,4 +167,66 @@ it("Returns 400 status code if client makes a request on a path that contains an
   })
 })
 
-})
\ No newline at end of file
+})
+
+describe("6) POST /api/articles/:article_id/comments", () => {
+it("Responds with 201 status code, adds new comment to database and sends the newly posted comment in the response.", () => {
+  return request(app)
+  .post('/api/articles/4/comments')
+  .send({username : "lurker", body: "To listen requires a voice, for what needs to be known requires us to ask."})
+  .expect(201)
+  .then(({body : {new_comment}}) => {
+     expect(new_comment).toEqual(
+      expect.objectContaining({
+          comment_id: 19,
+          body: "To listen requires a voice, for what needs to be known requires us to ask.",
+          article_id: 4,
+          author : "lurker",
+          votes: 0,
+          created_at: expect.any(String)
+      }));
+    });
+})
+  
+it("Returns 400 status code if request body is missing a required key.", () => {
+  return request(app)
+  .post('/api/articles/8/comments')
+  .send({body: "To listen requires a voice, for what needs to be known requires us to ask."})
+  .expect(400)
+  .then(({ body : { msg }}) => {
+    expect(msg).toBe("Bad request.");
+  });
+})
+  
+it("Returns 400 status code if username specified in request body does not reference an existing username in the users table (foreign key constraint violation).", () => {
+  return request(app)
+  .post('/api/articles/2/comments')
+  .send({username: "Racetrack", body: "Where did he go?"})
+  .expect(400)
```

Comment:
This would be 404 - username not found 
The user could exist, they just dont in the database.